src/lua/LuaSerializer.cpp

```diff
@@ -359,12 +367,85 @@ void LuaSerializer::UninitTableRefs()
 	lua_setfield(l, LUA_REGISTRYINDEX, "PiSerializer");
 
 	lua_pushnil(l);
-	lua_setfield(l, LUA_REGISTRYINDEX, "PiSerializerTableRefs");
+	lua_setfield(l, LUA_REGISTRYINDEX, NS_REFTABLE);
 
 	lua_pushnil(l);
 	lua_setfield(l, LUA_REGISTRYINDEX, "PiLuaRefLoadTable");
 }
 
+void LuaSerializer::SavePersistent(Json &json)
+{
+	lua_State *l = Lua::manager->GetLuaState();
+	LUA_DEBUG_START(l);
+
+	// NOTE: this must be an array for consistent deserialization order
+	Json persist = Json::array();
+
+	luaL_getsubtable(l, LUA_REGISTRYINDEX, NS_PERSISTENT);
+	lua_pushnil(l);
+
+	while (lua_next(l, -2)) {
+		Json entry = Json::object();
+
+		// persistent, id, value
+		std::string id = LuaPull<std::string>(l, -2);
+
+		// Serialize all registered persistent objects as a "backup" version
+		// (This will populate the PiSerializerTableRefs cache)
```

Comment:
This is just a comment, which I don't consider high priority to rewrite (if we mis-spell it as PiSierializerTablRefs the code will work correctly and not silently fail) but should probably be expanded to "This will populate the table identity cache for use in unserialization".